provider/oci/oci.go

```diff
@@ -169,11 +197,23 @@ func (p *OCIProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, error)
 	for _, zone := range zones {
 		var page *string
 		for {
-			resp, err := p.client.GetZoneRecords(ctx, dns.GetZoneRecordsRequest{
-				ZoneNameOrId:  zone.Id,
-				Page:          page,
-				CompartmentId: &p.cfg.CompartmentID,
-			})
+			// Get record based on zone scope
+			var resp dns.GetZoneRecordsResponse
+			var err error
+			if zone.Scope == "PRIVATE" {
```

Comment:
It seems like you making the API twice with and without `PRIVATE`, 
You could initiate the dns.GetZoneRecordsRequest before like

```
req := dns.GetZoneRecordsRequest{
		ZoneNameOrId:  zone.Id,
		Page:          page,
		CompartmentId: &p.cfg.CompartmentID
})

if zone.Scope == "Private" {
		req.Scope =  "PRIVATE"
}
resp, err = p.client.GetZoneRecords(ctx, req)
```

Reply:
Yes, previous implementation was kind of bad, hopefully it is better after refactoring. I think this can be resolved now.